metriq_gym/helpers/execution_time.py

```diff
@@ -0,0 +1,25 @@
+from functools import singledispatch
+
+from qbraid import QuantumJob
+from qbraid.runtime import QiskitJob, AzureQuantumJob
+from qiskit_ibm_runtime.execution_span import ExecutionSpans
+
+
+@singledispatch
```

Comment:
[praise]: I like this decorator pattern for consolidating the excitation time extraction.